Remove Options paradigm

dbook13 opened the pull request:
Per discussion [here](https://github.com/BoltApp/source/pull/10650#issuecomment-747713085), it no longer makes sense to be using the Options paradigm. This PR removes Options from both the auth request and refund request, and adds a field for the Last4 into the refund request.

nirajjayant reviewed (approved):
I think you can leave options in there for now even if we don't knowingly use it right now, but moving Last4 to being top level makes sense to me

dbook13 commented:
> I think you can leave options in there for now even if we don't knowingly use it right now, but moving Last4 to being top level makes sense to me

Added Options back into both requests and updated the PR title and description.
